### ISSUE TITLE
Unknown Error while setting up integration

### ISSUE THREAD

**Issue by ZeddicPrime** [eifinger/hass-here-weather#174]
## Version of the custom_component
2.1.5

## Configuration

```yaml

# Loads default set of integrations. Do not remove.
default_config:

# Load frontend themes from the themes folder
frontend:
  themes: !include_dir_merge_named themes

# Text to speech
tts:
  - platform: google_translate

python_script:

http:
  # Cloudflare settings
  use_x_forwarded_for: true
  trusted_proxies:
    - 172.30.33.0/24

# Zones
zone:
  - name: House
    latitude: # removed for privacy
    longitude: # removed for privacy
    radius: 7
    icon: mdi:home-city

# Templates
sensor:
  - platform: template
    sensors:
      victron_power_source_text:
        friendly_name: "Victron Power Source"
        value_template: >-
          {% set mapper =  {
            '0' : 'Unknown',
            '1' : 'Grid',
            '2' : 'Generator',
            '3' : 'Shore power',
            '240' : 'Not connected' } %}
          {% set state =  states.sensor.victron_power_source.state %}
          {{ mapper[state] if state in mapper else 'Unknown' }}
      googlefi_signal:
        friendly_name: "Google Fi Signal Strength"
        value_template: '{{ states("sensor.pep_wan_googlefi") }}'
        icon_template: >-
          {% set state = states('sensor.pep_wan_googlefi') %}
          {% if state == "4" %}
          mdi:signal-cellular-3
          {% elif state == "3" %}
          mdi:signal-cellular-2
          {% elif state == "2" %}
          mdi:signal-cellular-1
          {% elif state == "1" %}
          mdi:signal-cellular-outline
          {% else %}
          mdi:signal-off
          {% endif %}
      

binary_sensor:
  - platform: template
    sensors:
      starlink:
        friendly_name: Starlink
        device_class: connectivity
        value_template: '{{ is_state_attr("sensor.pep_wan_starlink", "statusLed", "green") }}'
  - platform: template
    sensors:
      housewifi:
        friendly_name: House Wi-Fi
        device_class: connectivity
        value_template: '{{ is_state_attr("sensor.pep_wan_home", "statusLed", "green") }}'
  - platform: template
    sensors:
      googlefi:
        friendly_name: Google Fi
        device_class: connectivity
        value_template: '{{ is_state_attr("sensor.pep_wan_googlefi", "statusLed", "green") }}'
  - platform: template
    sensors:
      otherwan:
        friendly_name: Misc WAN
        device_class: connectivity
        value_template: '{{ is_state_attr("sensor.pep_wan_other", "statusLed", "green") }}'
  - platform: template
    sensors:
      rvpluggedin:
        friendly_name: RV Plugged In
        device_class: plug
        value_template: '{{ is_state("sensor.victron_power_source", "3") }}'

rest_command:
  pep_cell_reset:
    url: "https://192.168.50.1/api/cmd.cellularModule.reset?accessToken={{ states('sensor.pep_token') }}"
    verify_ssl: false
    method: POST
    headers:
      Content-Type: application/json
    payload: '{"connId":"2"}'

automation: !include automations.yaml
script: !include scripts.yaml
scene: !include scenes.yaml
rest: !include rest.yaml
modbus: !include modbus.yaml

```

## Describe the bug
After installing the integration, restarting Home Assistant, then attempting to add the integration, after filling out the API key (left all other settings as default) and clicking Submit, the message "Unknown error orrcurred" is displayed and the setup does not proceed

## Debug log

```text

Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/aiohttp/web_protocol.py", line 435, in _handle_request
    resp = await request_handler(request)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/web_app.py", line 504, in _handle
    resp = await handler(request)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 60, in security_filter_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 222, in forwarded_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 28, in request_context_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 82, in ban_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 236, in auth_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/view.py", line 136, in handle
    result = await result
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 180, in post
    return await super().post(request, flow_id)
  File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 73, in wrapper
    result = await method(view, request, data, *args, **kwargs)
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 110, in post
    result = await self._flow_mgr.async_configure(flow_id, data)
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 280, in async_configure
    result = await self._async_handle_step(
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 367, in _async_handle_step
    result: FlowResult = await getattr(flow, method)(user_input)
  File "/config/custom_components/here_weather/config_flow.py", line 39, in async_step_user
    await async_validate_user_input(self.hass, user_input)
  File "/config/custom_components/here_weather/config_flow.py", line 20, in async_validate_user_input
    await here_client.weather_for_coordinates(
  File "/usr/local/lib/python3.10/site-packages/aiohere/aiohere.py", line 158, in weather_for_coordinates
    json_data = await self.request(params=params)
  File "/usr/local/lib/python3.10/site-packages/aiohere/aiohere.py", line 110, in request
    raise get_error_from_response(await response.json())
aiohere.exceptions.HereError: None

```

**Comment by eifinger** [comment]
Thank you for opening an issue and providing such detailed information.

Does this error occur several times at different times of day?

The HERE API can be "interesting" sometimes.

**Comment by ZeddicPrime** [comment]
I'll keep trying throughout the day. I only just signed up to get an API and I've tried a few times over the last 2 hours to set up the integration.  Also in the meantime I'll look at the API documentation and try to verify that I can use my API key and successfully get any data outside of HA

**Comment by ZeddicPrime** [comment]
It's still not working for me. I did find that some API tools work, and others do not. So right now my best guess is that something isn't set up right with my HERE account.
For example:
`curl -X GET -H 'Content-Type: *' --get 'https://weather.ls.hereapi.com/weather/1.0/report.json' --data-urlencode 'product=forecast_hourly' --data-urlencode 'name=Chicago' --data-urlencode 'apiKey=mykeyhere'`
This isn't working for me and returns `{"error":"Forbidden","error_description":"These credentials do not authorize access"}`. That curl command came straight from https://developer.here.com/documentation/examples/rest/auto_weather/weather-forecast-7days-hourly and does work with their example API key from the website.
I reviewed and made sure I followed the instructions at https://developer.here.com/tutorials/getting-here-credentials/ to the letter. I didn't turn on Trusted Domains or anything that could obviously limit my connections.

**Comment by eifinger** [comment]
I tried the curl command with my api key and it works.
I copied it from my account page under https://developer.here.com/projects:
![image](https://user-images.githubusercontent.com/1481961/201469406-f73fce85-88e6-43fd-ba48-c58fb55cf22d.png)

**Comment by ZeddicPrime** [comment]
When I log in to https://developer.here.com/projects it just says "To access your organization's projects and credentials please go here" with a link to https://platform.here.com/projects. And after I follow that link I don't actually see any place to get an API key.
The only place I can actually find a place to get an API key is to create an app under https://platform.here.com/admin/apps, and from there in the Credentials tab I can create API keys.

**Comment by eifinger** [comment]
Ah, they changed to different login methods. I currently don't have a HERE platform account but a developer account.

I will create one so I can replicate your setup.

**Comment by eifinger** [comment]
# tl;dr: They changed the API and this integration is currently broken.


I created a new "platform" account, created an api key and got the same result.

I then did the following:

1. When logged in, select "Projects Manager" from the top right corner
![image](https://user-images.githubusercontent.com/1481961/203731482-221d5ef2-b3c7-4e58-9624-e10d284f9353.png)
1. Create a new Project and select it or select an existing one (I already have my "old" one)
![image](https://user-images.githubusercontent.com/1481961/203731725-62a0e303-7ff8-4f6e-8edc-596612450515.png)
1. Under "Resources->Services" click "Link a Service"
![image](https://user-images.githubusercontent.com/1481961/203732143-498d871a-0a0b-44c5-9b64-db787463fb95.png)
1. Find "HERE Destination Weather" and click "Link"
![image](https://user-images.githubusercontent.com/1481961/203732310-00823ffe-7165-4aaf-8401-23688191b890.png)

And *THEN* I realized they changed the API.

But the new api key worked for me with the following query:
`curl -X GET -H 'Content-Type: *' --get 'https://weather.hereapi.com/v3/report' --data-urlencode 'products=forecastHourly' --data-urlencode 'q=Chicago' --data-urlencode 'apiKey=<API_KEY>'`

**Comment by eifinger** [comment]
It was a giant pain but the integration should work again. Please report any errors you might encounter =)
